lua/core/serial.lua

```diff
@@ -0,0 +1,250 @@
+--- USB serial device interface
+-- @module serial
+
+local serial = {
+    _handlers={},
+}
+
+--- add a new serial device handler
+function serial.handler(args)
+    assert(serial._handlers[args.id] == nil, "duplicate serial handler id: " .. args.id)
+    serial._handlers[args.id] = args
+end
+
+--- send a message to a serial device
+function serial.send(dev, line)
+    _norns.serial_send(dev, line)
+end
+
+_norns.serial = {}
+
+function _norns.serial.match(vendor, model, serial_num, interface_num)
+    for id, handler in pairs(serial._handlers) do
+        if handler.match(vendor, model, serial_num, interface_num) then
```

Comment:
I'm thinking about passing these attributes to `handler.match` in a readonly table instead of as positional arguments. That might make this a little easier to manage if attributes need to be added or removed in the future.